Cloneable/CloneableGenerator.cs

```diff
@@ -41,77 +76,131 @@ private void GenerateCloneMethods(GeneratorExecutionContext context)
             if (context.SyntaxReceiver is not SyntaxReceiver receiver)
                 return;
 
-            var classSymbols = GetClassSymbols(context, receiver);
+            var options = (context.Compilation as CSharpCompilation).SyntaxTrees[0].Options as CSharpParseOptions;
+            
+            var compilation = context.Compilation.AddSyntaxTrees(CSharpSyntaxTree.ParseText(SourceText.From(cloneableAttributeText, Encoding.UTF8), options)).
```

Comment:
This can be extracted as an extension method that gets called 3 times.

Reply:
Not done yet. Feel free to add it.